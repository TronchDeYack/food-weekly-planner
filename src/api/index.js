const recipeFake = {
  id: 'SPAGHETTI-ID',
  title: 'Spaghetti bolognaise',
  minutes: 20,
  caloriesLevel: 'medium',
  ingredients: [
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
  ],
};

const recipesFake = [
  recipeFake,
  {
    id: 'SOUPE-ID',
    title: 'Soupe à la tomate',
    minutes: 30,
    caloriesLevel: 'low',
    ingredients: [
      {
        id: 'TOMATE-ID',
        label: 'Tomates',
        quantity: 4,
        unit: 'UNIT',
        type: 'VEGETABLE',
      },
      {
        id: 'SALT-ID',
        label: 'Sel',
        quantity: 2,
        unit: 'CAC',
        type: 'CONDIMENT',
      },
    ],
  },
  {
    id: 'WELSH-ID',
    title: 'Welsh',
    minutes: 45,
    caloriesLevel: 'high',
    ingredients: [
      {
        id: 'CHEDDAR-ID',
        label: 'Cheddar',
        quantity: 400,
        unit: 'GR',
        type: 'DAIRY-PRODUCT',
      },
      {
        id: 'BEER-ID',
        label: 'Bière brune',
        quantity: 25,
        unit: 'CL',
        type: 'OTHER',
      },
      {
        id: 'HAM-ID',
        label: 'Jambon',
        quantity: 100,
        unit: 'GR',
        type: 'MEAT',
      },
      {
        id: 'BREAD-ID',
        label: 'Pain',
        quantity: 4,
        unit: 'SLICE',
        type: 'STARCHY-FOOD',
      },
    ],
  },
];

export function getRecipe(id) {
  console.log(`get recipe ${id}`);
  return Promise.resolve(recipeFake);
}

export function getRecipes() {
  return Promise.resolve(recipesFake);
}

export function getRecipesOfTheWeek() {
  return Promise.resolve([recipesFake[1], recipesFake[2]]);
}

export function getShoppingList() {
  return Promise.resolve([
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 1,
      unit: 'unit',
      type: 'STARCHY-FOOD',
      bought: false,
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
      bought: true,
    },
    {
      id: 'MILK-ID',
      label: 'Lait',
      quantity: 1,
      unit: 'L',
      type: 'DAIRY-PRODUCT',
      bought: false,
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
      bought: false,
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
      bought: true,
    },
    {
      id: 'CHICKEN-ID',
      label: 'Piou piou',
      quantity: 200,
      unit: 'GR',
      type: 'MEAT',
      bought: false,
    },
  ]);
}

export function getIngredients() {
  return Promise.resolve([
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'TOMATE-ID',
      label: 'Tomates',
      quantity: 4,
      unit: 'UNIT',
      type: 'VEGETABLE',
    },
    {
      id: 'SALT-ID',
      label: 'Sel',
      quantity: 2,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 4,
      unit: 'SLICE',
      type: 'STARCHY-FOOD',
    },
  ]);
}
