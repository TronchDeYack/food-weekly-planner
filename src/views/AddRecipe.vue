<template>
  <v-container>
    <v-stepper v-model="currentStep" vertical>
      <RecipeTitleStep :title="$t('addRecipe.steps.newTitle')"
        :step="1" :complete="currentStep > 1"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <MinuteStep :title="$t('addRecipe.steps.minutes')"
        :step="2" :complete="currentStep > 2"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <CaloriesLevelStep :title="$t('addRecipe.steps.calories')"
        :step="3" :complete="currentStep > 3"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.starchyFood')"
        :step="4" :complete="currentStep > 4"
        :ingredients="ingredients" ingredient-type="STARCHY-FOOD"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.vegetables')"
        :step="5" :complete="currentStep > 5"
        :ingredients="ingredients" ingredient-type="VEGETABLE"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.animalBased')"
        :step="6" :complete="currentStep > 6"
        :ingredients="ingredients" ingredient-type="MEAT"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.fruits')"
        :step="7" :complete="currentStep > 7"
        :ingredients="ingredients" ingredient-type="FRUIT"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.dairyProducts')"
        :step="8" :complete="currentStep > 8"
        :ingredients="ingredients" ingredient-type="DAIRY-PRODUCT"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.condiments')"
        :step="9" :complete="currentStep > 9"
        :ingredients="ingredients" ingredient-type="CONDIMENT"
        @next-step="updateStep" @previous-step="updateStep"
      />

      <IngredientStep :title="$t('addRecipe.steps.others')"
        :step="10" :complete="currentStep > 10"
        :ingredients="ingredients" ingredient-type="OTHER"
        @next-step="updateStep" @previous-step="updateStep"
      />
    </v-stepper>
  </v-container>
</template>

<script>
import { getIngredients } from '@/api';
import IngredientStep from '@/components/add-recipe/IngredientStep.vue';
import MinuteStep from '@/components/add-recipe/MinuteStep.vue';
import CaloriesLevelStep from '@/components/add-recipe/CaloriesLevelStep.vue';
import RecipeTitleStep from '@/components/add-recipe/RecipeTitleStep.vue';

export default {
  name: 'add-recipe',
  components: {
    CaloriesLevelStep,
    IngredientStep,
    MinuteStep,
    RecipeTitleStep,
  },
  data() {
    return {
      currentStep: 1,
      ingredients: [],
    };
  },
  mounted() {
    getIngredients().then((ingredients) => {
      this.ingredients = ingredients;
    });
  },
  // TODO: remove that by using a store
  methods: {
    updateStep({ step }) {
      this.currentStep = step;
    },
  },
};
</script>
