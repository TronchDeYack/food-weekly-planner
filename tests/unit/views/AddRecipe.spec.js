import AddRecipe from '@/views/AddRecipe.vue';
import { getIngredients } from '@/api';

import { createComponent } from '../../utils';

jest.mock('@/api');
describe('AddRecipe', () => {
  const ingredients = [
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      unit: 'GR',
      type: 'MEAT',
    },
  ];

  let wrapper;

  beforeEach(() => {
    getIngredients.mockResolvedValue(ingredients);
    wrapper = createComponent(AddRecipe);
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('state', () => {
    it('Should initialize "currentStep" to 1.', () => {
      wrapper = createComponent(AddRecipe);
      expect(wrapper.vm.$data.currentStep).toEqual(1);
    });

    it('Should initialize "ingredients" to an empty array.', () => {
      wrapper = createComponent(AddRecipe);
      expect(wrapper.vm.$data.ingredients).toEqual([]);
    });
  });

  describe('mounted', () => {
    it('Should get ingredients data from API and store them in "ingredients" state.', () => {
      expect(wrapper.vm.$data.ingredients).toEqual(ingredients);
    });
  });
});
