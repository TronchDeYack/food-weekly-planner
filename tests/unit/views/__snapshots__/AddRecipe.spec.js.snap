// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe Should render. 1`] = `
<v-container-stub tag="div">
  <v-stepper-stub value="1" vertical="true">
    <recipetitlestep-stub title="addRecipe.steps.newTitle" step="1"></recipetitlestep-stub>
    <minutestep-stub title="addRecipe.steps.minutes" step="2"></minutestep-stub>
    <calorieslevelstep-stub title="addRecipe.steps.calories" step="3"></calorieslevelstep-stub>
    <ingredientstep-stub title="addRecipe.steps.starchyFood" step="4" ingredients="[object Object],[object Object],[object Object]" ingredienttype="STARCHY-FOOD"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.vegetables" step="5" ingredients="[object Object],[object Object],[object Object]" ingredienttype="VEGETABLE"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.animalBased" step="6" ingredients="[object Object],[object Object],[object Object]" ingredienttype="MEAT"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.fruits" step="7" ingredients="[object Object],[object Object],[object Object]" ingredienttype="FRUIT"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.dairyProducts" step="8" ingredients="[object Object],[object Object],[object Object]" ingredienttype="DAIRY-PRODUCT"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.condiments" step="9" ingredients="[object Object],[object Object],[object Object]" ingredienttype="CONDIMENT"></ingredientstep-stub>
    <ingredientstep-stub title="addRecipe.steps.others" step="10" ingredients="[object Object],[object Object],[object Object]" ingredienttype="OTHER"></ingredientstep-stub>
  </v-stepper-stub>
</v-container-stub>
`;
